ebitenutil/loadembedimage.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 Hajime Hoshi
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.16
+// +build go1.16
+
+package ebitenutil
+
+import (
+	"embed"
+	"image"
+
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+// NewImageFromEmbedFile loads the file from embed.FS with path and returns ebiten.Image and image.Image.
+//
+// Image decoders must be imported when using NewImageFromEmbedFile. For example,
+// if you want to load a PNG image, you'd need to add `_ "image/png"` to the import section.
+func NewImageFromEmbedFile(embedFs embed.FS, path string) (*ebiten.Image, image.Image, error) {
```

Comment:
This should probably be a `fs.FS`. No need to restrict to `embed.FS`
I'd also name this `NewImageFromFileSystem`.

Reply:
That's absolutely true.